spond/spond.py

```diff
@@ -358,29 +356,39 @@ async def update_event(self, uid: str, updates: dict):
             "autoAccept": False,
             "payment": {},
             "attachments": [],
-            "id": None,
-            "tasks": {
-                "openTasks": [],
-                "assignedTasks": [
-                    {
-                        "name": None,
-                        "description": "",
-                        "type": "ASSIGNED",
-                        "id": None,
-                        "adultsOnly": True,
-                        "assignments": {"memberIds": [], "profiles": [], "remove": []},
-                    }
-                ],
-            },
+            "recipients": {"group": {"id": None}},
+            "tasks": {"openTasks": [], "assignedTasks": []},
         }
+        data = dict(base_event)
 
-        for key in base_event:
-            if event.get(key) is not None and not updates.get(key):
-                base_event[key] = event[key]
-            elif updates.get(key) is not None:
-                base_event[key] = updates[key]
+        if not self.events:
+            await self.get_events()
+        for event in self.events:
+            if event["id"] == uid:
+                data.update(event)
+                url = f"{self.api_url}sponds/{uid}"
+                break
+        else:
+            # No event of that id, create a new one (id to be set by Spond)
+            if (
+                len(updates.get("owners", [])) < 1
+                or updates["owners"][0].get("id") is None
+            ):
+                errmsg = '"owners" need to have a valid user id'
+                raise ValueError(errmsg)
```

Comment:
I suppose the second part should become a `LookupError` in the spirit of #135; for the first part, which covers an empty or altogether missing list of owners, it is not as clear – could of course handle both exceptions separately.
More or less the same for `recipients` below.

Reply:
> I suppose the second part should become a `LookupError` in the spirit of #135

Actually my error has confused things - as per discussion at #113, PR #135 was to change failed lookups to return `KeyError`, and I've now amended so it does.